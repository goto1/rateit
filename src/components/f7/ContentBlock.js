import React from "react";
import PropTypes from "prop-types";

const ContentBlock = ({ inset = false, className, children }) => {
  const classNames = `content-block ${inset ? "inset" : ""} ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

ContentBlock.propTypes = {
  inset: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default ContentBlock;
