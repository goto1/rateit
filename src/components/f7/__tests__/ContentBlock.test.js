import React from "react";
import ContentBlock from "../ContentBlock";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

describe("ContentBlock component", () => {
  const Content = <div>Lorem ipsum dolor sit amet</div>;

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ContentBlock>
          {Content}
        </ContentBlock>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with an `inset` style class", () => {
    const wrapper = shallow(
      <ContentBlock inset={true}>
        {Content}
      </ContentBlock>
    );
    expect(wrapper).toHaveClassName("inset");
  });

  it("renders without an `inset` style class", () => {
    const wrapper = shallow(
      <ContentBlock>
        {Content}
      </ContentBlock>
    );
    expect(wrapper).not.toHaveClassName("inset");
  });

  it("renders with a specific styling class", () => {
    const wrapper = shallow(
      <ContentBlock className="custom-class">
        {Content}
      </ContentBlock>
    );
    expect(wrapper).toHaveClassName("custom-class");
  });
});
