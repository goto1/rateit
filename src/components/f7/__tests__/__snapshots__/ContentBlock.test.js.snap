// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlock component renders correctly 1`] = `
<div
  className="content-block  "
>
  <div>
    Lorem ipsum dolor sit amet
  </div>
</div>
`;
